crates/miden-lib/src/transaction/inputs.rs

```diff
@@ -1,52 +1,139 @@
 use alloc::vec::Vec;
 
 use miden_objects::account::{AccountHeader, AccountId, PartialAccount};
-use miden_objects::block::AccountWitness;
+use miden_objects::block::{AccountWitness, BlockHeader, BlockNumber};
 use miden_objects::crypto::SequentialCommit;
 use miden_objects::crypto::merkle::InnerNodeInfo;
+use miden_objects::note::{Note, NoteInclusionProof};
 use miden_objects::transaction::{
     AccountInputs,
     InputNote,
+    InputNotes,
     PartialBlockchain,
     TransactionArgs,
-    TransactionInputs,
+    TransactionPreparationInputs,
+    TransactionScript,
 };
 use miden_objects::vm::AdviceInputs;
-use miden_objects::{EMPTY_WORD, Felt, FieldElement, Word, ZERO};
-use miden_processor::AdviceMutation;
+use miden_objects::{EMPTY_WORD, Felt, FieldElement, TransactionInputError, Word, ZERO};
+use miden_processor::{AdviceMutation, StackInputs};
 use thiserror::Error;
 
 use super::TransactionKernel;
 
-// TRANSACTION ADVICE INPUTS
+// TRANSACTION KERNEL INPUTS
 // ================================================================================================
 
-/// Advice inputs wrapper for inputs that are meant to be used exclusively in the transaction
-/// kernel.
-#[derive(Debug, Clone, Default)]
-pub struct TransactionAdviceInputs(AdviceInputs);
+/// Contains the data required to execute a transaction.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TransactionKernelInputs {
+    prep_inputs: TransactionPreparationInputs,
+    input_notes: InputNotes<InputNote>,
+    tx_args: TransactionArgs,
+    init_advice_inputs: Option<AdviceInputs>,
+}
```

Comment:
I would rename `init_advice_inputs` into just `advice_inputs`. Also, not sure if this field needs to be optional. We could always initialize it with empty advice inputs.